stl/inc/algorithm

```diff
@@ -7594,9 +7599,78 @@ namespace ranges {
         }
     }
     // clang-format on
+
+    // VARIABLE ranges::sort
+    class _Sort_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <random_access_iterator _It, sentinel_for<_It> _Se, class _Pr = ranges::less, class _Pj = identity>
+            requires sortable<_It, _Pr, _Pj>
+        _NODISCARD constexpr _It operator()(_It _First, _Se _Last, _Pr _Pred = {}, _Pj _Proj = {}) const {
+            // clang-format on
+            _Adl_verify_range(_First, _Last);
+            auto _UFirst = _Get_unwrapped(_STD move(_First));
+            auto _ULast  = _Get_final_iterator_unwrapped<_It>(_UFirst, _STD move(_Last));
+            _Seek_wrapped(_First, _ULast);
+            const auto _Count = _ULast - _UFirst;
+            _Sort_common(_STD move(_UFirst), _STD move(_ULast), _Count, _Pass_fn(_Pred), _Pass_fn(_Proj));
+            return _First;
+        }
+
+        // clang-format off
+        template <random_access_range _Rng, class _Pr = ranges::less, class _Pj = identity>
+            requires sortable<iterator_t<_Rng>, _Pr, _Pj>
+        _NODISCARD constexpr borrowed_iterator_t<_Rng> operator()(_Rng&& _Range, _Pr _Pred = {}, _Pj _Proj = {}) const {
+            // clang-format on
+            auto _ULast       = _Get_final_iterator_unwrapped(_Range);
+            auto _UFirst      = _Ubegin(_Range);

```

Comment:
Style: It's a little strange to see `_ULast` defined before `_UFirst`.

Reply:
This is an attempt to help the compiler avoid spilling `_UFirst` into memory while evaluating `_Get_final_iterator_unwrapped(_Range)`, which is a bit silly given that I've never so much as looked at codegen for any specialization of this template. I'll reorder these.